paddleseg/models/pfpnnet.py

```diff
@@ -0,0 +1,205 @@
+                             enable_auxiliary_loss=self.enable_auxiliary_loss)
+        self.init_weight()
+
+    def forward(self, x):
+        feats = self.backbone(x)
+        feats = [feats[i] for i in self.backbone_indices]
+        logit_list = self.head(feats)
+        return [
+            F.interpolate(logit,
+                          paddle.shape(x)[2:],
+                          mode='bilinear',
+                          align_corners=self.align_corners)
+            for logit in logit_list
+        ]
+
+    def init_weight(self):
+        if self.pretrained is not None:
+            utils.load_entire_model(self, self.pretrained)
+
+
+class PFPNHead(nn.Layer):
+    """
+    The PFPNHead implementation.
+
+    Args:
+        inplane (int): Input channels of PPM module.
+        num_class (int): The unique number of target classes.
+        fpn_inplanes (list): The feature channels from backbone.
+        fpn_dim (int, optional): The input channels of FPN module. Default: 512.
+        enable_auxiliary_loss (bool, optional): A bool value indicates whether adding auxiliary loss. Default: False.
+    """
+    def __init__(self,
+                 num_class,
+                 fpn_inplanes,
+                 channels,
+                 dropout_ratio=0.1,
+                 fpn_dim=256,
+                 enable_auxiliary_loss=False,
+                 align_corners=False):
+        super(PFPNHead, self).__init__()
+        self.enable_auxiliary_loss = enable_auxiliary_loss
+        self.align_corners = align_corners
+        self.lateral_convs = []
+        self.fpn_out = []
+
+        for fpn_inplane in fpn_inplanes:
+            self.lateral_convs.append(
+                nn.Sequential(nn.Conv2D(fpn_inplane, fpn_dim, 1),
+                              layers.SyncBatchNorm(fpn_dim), nn.ReLU()))
+            self.fpn_out.append(
+                nn.Sequential(
+                    layers.ConvBNReLU(fpn_dim, fpn_dim, 3, bias_attr=False)))
+
+        self.lateral_convs = nn.LayerList(self.lateral_convs)
+        self.fpn_out = nn.LayerList(self.fpn_out)
+
+        self.scale_heads = []
+        for index in range(len(fpn_inplanes)):
+            head_length = max(
+                1, int(np.log2(fpn_inplanes[index]) - np.log2(fpn_inplanes[0])))
+            scale_head = []
+            for head_index in range(head_length):
+                scale_head.append(
+                    layers.ConvBNReLU(
+                        fpn_dim,
+                        channels,
+                        3,
+                        padding=1,
+                    ))
+                if fpn_inplanes[index] != fpn_inplanes[0]:
+                    scale_head.append(
+                        nn.Upsample(scale_factor=2,
+                                    mode='bilinear',
+                                    align_corners=align_corners))
+            self.scale_heads.append(nn.Sequential(*scale_head))
+        self.scale_heads = nn.LayerList(self.scale_heads)
```

Comment:
self.scale_head can be simplified referring to the previous comment.

Reply:
fix.